### PR TITLE
feat(cli): add selector and field-selector option to terminate

### PR DESCRIPTION
**Pull request by hanjunlee**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 

I've added two options, `selector` and `field-selector`, to the `terminate` command. And this PR is separated from #3962 as Alexec recommended. It's welcome if you advise me. @simster7

### CODE REVIEW

File: cmd/argo/commands/terminate.go [hunk 1]
```diff
@@ -22,19 +54,47 @@ func NewTerminateCommand() *cobra.Command {
   argo terminate @latest
 `,
 		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) != 0 && t.isList() {
```
Comment (alexec) [inline cmd/argo/commands/terminate.go:57]: xargs friendly?

File: cmd/argo/commands/terminate.go [hunk 2]
```diff
@@ -22,19 +54,47 @@ func NewTerminateCommand() *cobra.Command {
...
 				})
 				errors.CheckError(err)
 				fmt.Printf("workflow %s terminated\n", wf.Name)
 			}
 		},
 	}
+
+	command.Flags().StringVarP(&t.labels, "selector", "l", "", "Selector (label query) to filter on, not including uninitialized ones")
+	command.Flags().StringVar(&t.fields, "field-selector", "", "Selector (field query) to filter on, supports '=', '==', and '!='.(e.g. --field-selectorkey1=value1,key2=value2). The server only supports a limited number of field queries per type.")
+	command.Flags().BoolVar(&t.dryRun, "dry-run", false, "Do not delete the workflow, only print what would happen")
```
Comment (alexec) [inline cmd/argo/commands/terminate.go:98]: `Do not terminate`?

File: cmd/argo/commands/terminate.go [hunk 3]
```diff
@@ -22,19 +54,47 @@ func NewTerminateCommand() *cobra.Command {
...
+					labels:    t.labels,
+				})
+				errors.CheckError(err)
+				workflows = append(workflows, listed...)
+			} else {
+				workflows = t.convertToWorkflows(args)
+			}
+
+			for _, w := range workflows {
+				if t.dryRun {
```
Comment (alexec) [inline cmd/argo/commands/terminate.go:81]: dry-run test?

**Review by alexec — CHANGES_REQUESTED** [review]
Changes requested

File: test/e2e/cli_test.go [hunk 4]
```diff
@@ -692,6 +692,19 @@ func (s *CLISuite) TestWorkflowTerminate() {
 		})
 }
 
+func (s *CLIWithServerSuite) TestWorkflowTerminateDryRun() {
+	s.testNeedsOffloading()
```
Comment (alexec) [inline test/e2e/cli_test.go]: I don't think this needs offloading?

**Review by alexec — APPROVED** [review]

File: test/e2e/cli_test.go (hunk 4 above)
Comment (hanjunlee) [reply test/e2e/cli_test.go]: It was already fixed ~

**Pull request merged**